### PR TITLE
Russian Peasant Algorithm

### PR DESCRIPTION
**Pull request by narayan2111**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Russian Peasant algorithm to multiply two long numbers.

### Checklist

- [Y ] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#978

### CODE REVIEW

**Comment by narayan2111** [comment]
Kindly Review This PR
@plazzy99 @iamrajiv @Abhijit2505 @Kajol-Kumari

**Review by vatsalkesarwani12 — APPROVED** [review]

File: Python/other/Russian_Peasant_Algorithm.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	# While second number doesn't 
+	# become 1 
+	while (number2 > 0): 
+	
+		# If second number becomes odd, add the first number to result 
+		if (number2 & 1): 
+			result = result + number1 
+
+		# Double the first number and halve the second number 
+		number1 = number1 << 1 		 #bitwise left operator
```
Comment (atarax665) [inline Python/other/Russian_Peasant_Algorithm.py]:
```suggestion
		number1 = number1 << 1 		 #bitwise left shift operator
```

File: Python/other/Russian_Peasant_Algorithm.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		number2 = number2 >> 1 		 #bitwise right operator
```
Comment (atarax665) [inline Python/other/Russian_Peasant_Algorithm.py]:
```suggestion
		number2 = number2 >> 1 		 #bitwise right shift operator
```

File: Python/other/Russian_Peasant_Algorithm.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	
+	return print("Result is",result )
+
+# Driver program to test
```
Comment (atarax665) [inline Python/other/Russian_Peasant_Algorithm.py]:
```suggestion
# Driver code 
```

File: Python/other/Russian_Peasant_Algorithm.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+			result = result + number1 
+
+		# Double the first number and halve the second number 
+		number1 = number1 << 1 		 #bitwise left operator
+		number2 = number2 >> 1 		 #bitwise right operator
+	
+	return print("Result is",result )
+
+# Driver program to test 
+# above function
```
Comment (atarax665) [inline Python/other/Russian_Peasant_Algorithm.py]:
```suggestion
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Python/other/Russian_Peasant_Algorithm.py (hunk 4 above)
Comment (narayan2111) [reply Python/other/Russian_Peasant_Algorithm.py]: > 

Updated the code  @atarax665

**Comment by narayan2111** [comment]
kindly review this PR @iamrajiv

**Review by iamrajiv — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Comment by narayan2111** [comment]
kindly merge this PR it has 3 reviews @iamrajiv

**Pull request merged**